tests/main/install-hook/task.yaml

```diff
@@ -0,0 +1,48 @@
+summary: Check install and remove hooks.
+
+execute: |
+    . $TESTSLIB/snaps.sh
+    install_local snap-hooks
+
+    echo "Verify configuration value with snap get"
+    snap get snap-hooks installed | MATCH 1
+    snap get snap-hooks foo | MATCH bar
+
+    echo "Verify that install hook is run only once"
+    snap set snap-hooks installed=2
+    install_local snap-hooks
+    snap get snap-hooks installed | MATCH 2
+
+    REMOVE_HOOK_FILE=/root/snap/snap-hooks/current/remove-hook-executed
+
+    echo "Verify that remove hook is not executed when removing single revision"
+    snap remove --revision=x1 snap-hooks
+    if test -f $REMOVE_HOOK_FILE; then
+        echo "Remove hook was executed. It shouldn't."
+        exit 1
+    fi
+
+    echo "Verify that remove hook is executed"
+    snap remove snap-hooks
+    if ! test -f $REMOVE_HOOK_FILE; then
```

Comment:
Ok, as discussed /root/snap/<snapname> is currently not cleaned on removal because we only clean "/home/*/snap/..." (it's probably a bug). I've changed the test to drop file in home directory instead so that the test doesn't rely on misbehavior that can soon change if we fix that problem.